pkg/controller/infrastructure/controller.go

```diff
@@ -77,24 +79,31 @@ func DefaultPredicates(client client.Client, typeName string, ignoreOperationAnn
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager, args AddArgs) error {
 	args.ControllerOptions.Reconciler = NewReconciler(mgr, args.Actuator)
-	return add(mgr, args.ControllerOptions, args.Predicates, args.WantsClusterWatch)
+	return add(mgr, args)
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, options controller.Options, predicates []predicate.Predicate, clusterWatch bool) error {
-	ctrl, err := controller.New(ControllerName, mgr, options)
+func add(mgr manager.Manager, args AddArgs) error {
+	ctrl, err := controller.New(ControllerName, mgr, args.ControllerOptions)
 	if err != nil {
 		return err
 	}
 
-	if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Infrastructure{}}, &handler.EnqueueRequestForObject{}, predicates...); err != nil {
+	if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Infrastructure{}}, &handler.EnqueueRequestForObject{}, args.Predicates...); err != nil {
 		return err
 	}
-	if err := ctrl.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: SecretToInfrastructureMapper(mgr.GetClient(), predicates)}); err != nil {
+	if err := ctrl.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: SecretToInfrastructureMapper(mgr.GetClient(), args.Predicates)}); err != nil {
 		return err
 	}
-	if clusterWatch {
-		if err := ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.Cluster{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: ClusterToInfrastructureMapper(mgr.GetClient(), predicates)}); err != nil {
+	if args.WantsClusterWatch {
```

Comment:
Now that we can enable adding any arbitrary watch we probably can remove this `WantsClusterWatch`. Any actuator that wants to have this can simply use the `WatchBuilder` to register this. WDYT?

Reply:
You are absolutely right. I was still wondering if this argument has not been used by some providers.
However, I cannot see any provider from gardener-extensions that uses it. In this case, I will update the PR.